Cron, simple postgres backups

rollcat opened the pull request:
Two new images:

https://hub.docker.com/r/unit9/cron/
https://hub.docker.com/r/unit9/postgres-dumper/

- `unit9/cron` is a shell script hack that pretends to be cron. The reason for not running the real thing, is that absolutely EVERYTHING about (Debian's) cron makes it Docker-unfriendly and extraordinarily painful:

  - It insists on logging stuff to syslog (making debugging itself extraordinarily painful)
  - It insists on sending the job output via email (making debugging faulty jobs extraordinarily painful)
  - It resets the environment for each job (making configuring things extraordinarily painful)

  So I gave up, wrote fifty lines of shell, and it Just Works™ now.

- `unit9/postgres-dumper` uses :point_up: to run a daily `pg_dumpall | gzip` and some `gsutil` to upload that to GCS. There's no garbage collection = infinite retention.

rollcat committed:
[cron] Add an image for running cron(8) jobs

rollcat committed:
[postgres-dumper] WIP

rollcat committed:
[postgres-dumper] GCS integration

rollcat committed:
[postgres-dumper] Add BOTO_PATH

rollcat committed:
[postgres-dumper] Don't create empty files

In case connection failed!

rollcat committed:
[wip] cron!

rollcat committed:
[cron] rewrite cron with a shell script

Seriously, cron is the most Docker-hostile thing I've ever witnessed.

rollcat committed:
[postgres-dumper] GOOGLE_APPLICATION_CREDENTIALS

esenti commented on cron/readme.md:
```diff
@@ -0,0 +1,20 @@
+
+- `/etc/cron.hourly` - Every hour at 17th minute
+- `/etc/cron.daily` - Every 24 hours at 6:25 AM
+- `/etc/cron.weekly` - Every Saturday at 6:47 AM
```
Where are weekly and monthly schedules implemented?

rollcat replied:
See note on https://github.com/unit9/docklabs/pull/25/commits/3b1e442a3c0f4c44d4dac2d5b3149846e412e381

Perhaps better not to implement these at all

rollcat committed:
[fix] WHOOPS fix rc.local

rollcat committed:
[cron] weekly and monthly are not implemented

Also, not sure if having that in a stateless container makes any sense.
Perhaps anacron with a persistent spool dir would work better.
